plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c

```diff
@@ -885,8 +885,22 @@ static bool set_config(struct ffmpeg_output *stream)
 	config.format_mime_type = "video/M2PT";
 
 	/* 2. video settings */
-	// 2.a) set video format from obs to FFmpeg
-	video_t *video = obs_output_video(stream->output);
+
+	// 2.a) set width & height
+	config.width = (int)obs_output_get_width(stream->output);
+	config.height = (int)obs_output_get_height(stream->output);
+	config.scale_width = config.width;
+	config.scale_height = config.height;
+
+	// 2.b) set video codec & id from video encoder
+	obs_encoder_t *vencoder = obs_output_get_video_encoder(stream->output);
+	config.video_encoder = obs_encoder_get_codec(vencoder);
+	if (strcmp(config.video_encoder, "h264") == 0)
+		config.video_encoder_id = AV_CODEC_ID_H264;
+	else
+		config.video_encoder_id = AV_CODEC_ID_AV1;
+	// 2.c) set video format from obs to FFmpeg
```

Comment:
Nit: please always leave a blank line after an unenclosed `if`, `if-else`, or `if-else if-else` statement to make it clear that we have exited that statement.

Reply:
done